src/extension/settings/settingsPropertyManager.ts

```diff
@@ -108,17 +108,28 @@ export class SettingsPropertyManager {
             profileObject = general.plainObjectToClass(ConanProfileConfiguration, selectedProfileObject);
         }
 
+        let pythonInterpreter = await python.getCurrentPythonInterpreter();
+        if (pythonInterpreter !== undefined) {
```

Comment:
Need an `if` statement before entering this section, since the `profileObject` might be `undefined`, it leads to error in case of the `undefined` object, when you want to set the value to its attribute.

Reply:
I was expecting the same output as in the explanation but apparently the it didn't do the things as expected.
`stack trace: TypeError: Cannot set properties of undefined (setting 'conanPythonInterpreter')`

I would suggest to remove the `!` and `?` for assertion and surround the whole code with simple `if` condition before entering the section of entering the python interpreter, see code below. With this we can make sure that it will return `undefined` in case the object doesn't exist

```typescript
.
.
if(profileObject) {
    let pythonInterpreter = await python.getCurrentPythonInterpreter();
    if (pythonInterpreter !== undefined) {
        ...
    }
}

return profileObject
```

Reply:
I was not smart enough :P The `if (!` was the problem. `profileObject?` evaluates to `false` so `if (!profileObject?...)` is `true` 🤦‍♂️